pypeit/coadd3d.py

```diff
@@ -0,0 +1,1772 @@
+                image - Use the relative scale that was derived from the science frame
+                none - Do not perform relative scale correction
+
+        Returns:
+            :obj:`str`_: A string that describes the scale correction mode to be used (see opts_scalecorr description)
+            `numpy.ndarray`_: 2D image (same shape as science frame) containing the relative spectral scaling to apply to the science frame
+        """
+        this_scalecorr = self.scalecorr_default
+        relScaleImg = self.relScaleImgDef.copy()
+        if opts_scalecorr is not None:
+            if opts_scalecorr.lower() == 'default':
+                if self.scalecorr_default == "image":
+                    relScaleImg = spec2DObj.scaleimg
+                    this_scalecorr = "image"  # Use the current spec2d for the relative spectral illumination scaling
+                else:
+                    this_scalecorr = self.scalecorr_default  # Use the default value for the scale correction
+            elif opts_scalecorr.lower() == 'image':
+                relScaleImg = spec2DObj.scaleimg
+                this_scalecorr = "image"  # Use the current spec2d for the relative spectral illumination scaling
+            elif opts_scalecorr.lower() == 'none':
+                relScaleImg = np.array([1])
+                this_scalecorr = "none"  # Don't do relative spectral illumination scaling
+            else:
+                # Load a user specified frame for sky subtraction
+                msgs.info("Loading the following frame for the relative spectral illumination correction:" +
+                          msgs.newline() + opts_scalecorr)
+                try:
+                    spec2DObj_scl = spec2dobj.Spec2DObj.from_file(opts_scalecorr, self.detname)
+                except:
+                    msgs.error(
+                        "Could not load skysub image from spec2d file:" + msgs.newline() + opts_scalecorr)
+                relScaleImg = spec2DObj_scl.scaleimg
+                this_scalecorr = opts_scalecorr
+        if this_scalecorr == "none":
+            msgs.info("Relative spectral illumination correction will not be performed.")
+        else:
+            msgs.info("Using the following frame for the relative spectral illumination correction:" +
+                      msgs.newline() + this_scalecorr)
+        # Return the scaling correction for this frame
+        return this_scalecorr, relScaleImg
+
+    def set_default_skysub(self):
+        """
+        Set the default mode to use for sky subtraction.
+        """
+        if self.cubepar['skysub_frame'] in [None, 'none', '', 'None']:
+            self.skysub_default = "none"
+            self.skyImgDef = np.array([0.0])  # Do not perform sky subtraction
+            self.skySclDef = np.array([0.0])  # Do not perform sky subtraction
+        elif self.cubepar['skysub_frame'].lower() == "image":
```

Comment:
These .lower() instances are not necessary. We have functionality in the parsets to only tolerate specific options, and the codes fault when those options are not passed in.

Reply:
OK - I'll move this to the parset

Reply:
Right... the issue here is that the it uses the parset, and then there's also a chance to override the default setting with `pypeit.inputfiles.Coadd3DFile()`. In the case of the `default` setting, we can remove the `.lower()`, and I've done that now, but I think the issue is that there is a conflict in the way we handle `inputfiles.py` vs the parset.